tr2/lara: add configurable HP and perma-damage options

lahm86 opened the pull request:
Resolves #3179.
Resolves #3180.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

Adds TR1's HP and perma-damage options to TR2, and ensures demos in both games have max HP to allow playing out normally.

`Lara_Initialise` is slightly different in TR2 to cater for cutscenes, as these would set her health back to the maximum, and as they don't have resume info, this would mean that would carry forward to the next level. It seems to check out for me skipping through the levels normally on 1HP, but I'd appreciate a sanity check here.

lahm86 committed:
tr1/demo: reset Lara's HP in demos

This ensures Lara's HP is at the maximum for demos to allow them to
play out normally.

Resolves #3180.

lahm86 committed:
tr2/lara: add configurable HP and perma-damage options

This adds options to TR2 to define Lara's starting HP and to allow
toggling healing between levels.

Resolves  #3179.

rr- commented on src/libtrx/config/priv_tr2.c:
```diff
@@ -185,4 +186,5 @@ void Config_Sanitize(void)
     CLAMP(g_Config.visuals.fov, 30, 150);
     CLAMP(g_Config.ui.bar_scale, 0.5, 2.0);
     CLAMP(g_Config.ui.text_scale, 0.5, 2.0);
+    CLAMP(g_Config.gameplay.start_lara_hitpoints, 1, LARA_MAX_HITPOINTS);
```
IIRC we have SanitizeCommon that can handle duplicate code now.

lahm86 replied:
Thank you. I've added a separate commit for this, was able to shift quite a few things over.

lahm86 committed:
config/priv: move common sanitization calls

This moves common calls for sanitzing config values to the main config
priv module to reduce duplication.
